modules/collect/src/main/java/com/opengamma/strata/collect/io/ArrayByteSource.java

```diff
@@ -149,21 +155,32 @@ public static ArrayByteSource ofUtf8(String str) {
 
   //-------------------------------------------------------------------------
   /**
-   * Creates an instance from another byte source.
+   * Obtains an instance from another byte source.
    * 
    * @param other  the other byte source
    * @return the byte source
    * @throws UncheckedIOException if an IO error occurs
    */
   public static ArrayByteSource from(ByteSource other) {
-    if (other instanceof ArrayByteSource) {
-      return (ArrayByteSource) other;
+    if (other instanceof BeanByteSource) {
+      return ((BeanByteSource) other).load();
```

Comment:
This worries me - it looks like the default implementation calls this `from` method, so if somebody forgets to override it (or hasn't yet) then we get infinite recursion and a stack overflow

Reply:
And why get rid of the fast path without copying?